tests/cnf/core/network/sriov/tests/qinq.go

```diff
@@ -885,3 +1065,133 @@ func runQinQDpdkTestCases(nodeName, serverName, clientName, sriovNetworkName, na
+		APIClient,
+		vfioPCIName,
+		NetConfig.SriovOperatorNamespace,
+		vfioPCIResName,
+		16,
+		[]string{fmt.Sprintf("%s#10-15", sriovInterface)},
+		NetConfig.WorkerLabelMap).WithDevType("vfio-pci")
+	if sriovDeviceID == netparam.MlxDeviceID {
+		_, err = sriovPolicy.WithRDMA(true).Create()
+	} else {
+		_, err = sriovPolicy.Create()
+	}
+
+	Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("Failed to create sriovnetwork policy %s",
+		vfioPCIName))
+
+	By("Waiting until cluster MCP and SR-IOV are stable")
+
+	err = netenv.WaitForSriovAndMCPStable(
+		APIClient, tsparams.MCOWaitTimeout, time.Minute, NetConfig.CnfMcpLabel, NetConfig.SriovOperatorNamespace)
+	Expect(err).ToNot(HaveOccurred(), "Failed cluster is not stable")
+}
+
+func defineAndCreateSriovNetworks(sriovNetworkPromiscName, sriovNetworkDot1ADName, sriovNetworkDot1QName,
+	sriovResName string) {
+	vlan, err := strconv.Atoi(NetConfig.VLAN)
+	fmt.Println("VLAN", vlan)
+	Expect(err).ToNot(HaveOccurred(), "Failed to convert VLAN value")
+
+	By("Define and create sriov-network for the promiscuous client")
+
+	_, err = sriov.NewNetworkBuilder(APIClient,
+		sriovNetworkPromiscName, NetConfig.SriovOperatorNamespace, tsparams.TestNamespaceName,
+		sriovResName).WithTrustFlag(true).Create()
+	Expect(err).ToNot(HaveOccurred(),
+		fmt.Sprintf("Failed to create sriov network srIovNetworkPromiscuous %s", err))
+
+	By("Define and create sriov-network with 802.1ad S-VLAN")
+	defineAndCreateSrIovNetworkWithQinQ(sriovNetworkDot1ADName, sriovResName, "802.1ad",
+		uint16(vlan))
+	Expect(err).ToNot(HaveOccurred(),
+		fmt.Sprintf("Failed to create sriov network srIovNetworkDot1AD %s", err))
+
+	By("Define and create sriov-network with 802.1q S-VLAN")
+	defineAndCreateSrIovNetworkWithQinQ(sriovNetworkDot1QName, sriovResName, "802.1q",
+		uint16(vlan))
+	Expect(err).ToNot(HaveOccurred(),
+		fmt.Sprintf("Failed to create sriov network srIovNetworkDot1Q %s", err))
+}
+
+func defineAndCreateNAD(nadCVLAN100, nadCVLAN101, nadMasterBond0, intNet1 string) {
+	By("Define and create a network attachment definition with a C-VLAN 100")
+
+	_, err := define.VlanNad(APIClient, nadCVLAN100, tsparams.TestNamespaceName, "net1", 100,
+		nad.IPAMStatic())
+	Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("Fail to create Network-Attachment-Definition %s",
+		nadCVLAN100))
+
+	By("Define and create a network attachment definition with a C-VLAN 101")
+
+	_, err = define.VlanNad(APIClient, nadCVLAN101, tsparams.TestNamespaceName, intNet1, 101,
+		nad.IPAMStatic())
+	Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("Fail to create Network-Attachment-Definition %s",
+		nadCVLAN101))
+
+	By("Define and create a Bonded network attachment definition with a C-VLAN 100")
+
+	bondMasterNad := defineQinQBondNAD(nadMasterBond0, "active-backup")
+	_, err = bondMasterNad.Create()
+	Expect(err).ToNot(HaveOccurred(), fmt.Sprintf("Fail to create a Bond Network-Attachment-Definition %s",
+		nadCVLAN100))
+}
+
+func removeSriovConfigurations() {
```

Comment:
Please use sriovenv.RemoveSriovConfigurationAndWaitForSriovAndMCPStable and use this func in your after all

Reply:
Changed